### PR TITLE
Mark Gc.create_alarm as [@inline never]

### PR DESCRIPTION
**Pull request by poechsel**
Flambda2 is able to lift allocations happening from inside `Gc.create_alarm` to the toplevel. This is problematic has `finalise` can't work with statically allocated objects.

### CODE REVIEW

File: stdlib/gc.ml [hunk 1]
```diff
@@ -113,7 +113,11 @@ let rec call_alarm arec =
   end
 
 
-let create_alarm f =
+(* [create_alarm] are marked as [@inline never] to prohibit the inliner from
+   lifting to toplevel any allocations happening inside it. Flambda2 has a
+   tendency to lift [arec] and to statically allocate it, which is incompatible
+   with the semantics of [finalise]. *)
```
Comment (mshinwell) [inline stdlib/gc.ml]: This is a bit wordy.  How about:

```
We use [@inline never] to ensure [arec] is never statically allocated (which would prevent installation of the finaliser).
```

**Review by mshinwell — CHANGES_REQUESTED** [review]

File: stdlib/gc.ml (hunk 1 above)
Comment (poechsel) [reply stdlib/gc.ml]: It's better!

**Review by mshinwell — APPROVED** [review]

**Comment by mshinwell** [comment]
OK to merge if CI passes.
